### ISSUE TITLE
Pi-hole Installation breaks previous OpenVPN setup

### ISSUE THREAD

**Issue by Clubguy** [pi-hole/pi-hole#1487]
**In raising this issue, I confirm the following (please check boxes, eg [X]) Failure to fill the template will close your issue:**

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*
- [X] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the codebase?:**

_3_

---
**[FEATURE REQUEST | QUESTION | OTHER]:**

Please [submit your feature request here](https://discourse.pi-hole.net/c/feature-requests), so it is votable by the community.  It's also easier for us to track.

**[BUG | ISSUE] Expected Behaviour:**
Pi-hole will not break pre-existing OpenVPN system already in place

**[BUG | ISSUE] Actual Behaviour:**
OpenVPN no longer works after installation of Pi-hole

**[BUG | ISSUE] Steps to reproduce:**
Devices cannot connect to OpenVPN on the internal network nor the external network
-
-
-
-

**(Optional) Debug token generated by `pihole -d`:**

`<token>`

_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

**Comment by Clubguy** [comment]
I read the similar article in the closed cases section but that did not work for my installation.  Debug token is:

ai222ygm5f

Thank you

**Comment by technicalpyro** [comment]
if you are willing please provide (with redacted certificates and ip's)use place holders like ]EXTERNAL IP so we know though. the server conf files anf or logs.  the description of your issue leaves out a lot of info that would help sort this out

**Comment by Clubguy** [comment]
Ok, my webserver is showing the default Apache page rather than the hosted web page too.  Let's start with that config file:  

mydomain.com.conf
<VirtualHost *:8080>  #This was set to port 80 before the Pi-hole installation.  Both ports are open on the #firewall
ServerName www.mydomain.com
ServerAlias mydomain.com *.mydomain.com
DocumentRoot /var/www/mydomain.com/public_html
</VirtualHost>

Here is the server config file:
server.conf
local 192.168.0.104 # SWAP THIS NUMBER WITH YOUR RASPBERRY PI IP ADDRESS
dev tun
proto tcp  #Some people prefer to use tcp. Don't change it if you don't know.
port 1194
ca /etc/openvpn/easy-rsa/keys/ca.crt
cert /etc/openvpn/easy-rsa/keys/mydomain.com.crt # SWAP WITH YOUR CRT NAME
key /etc/openvpn/easy-rsa/keys/mydomain.com.key # SWAP WITH YOUR KEY NAME
dh /etc/openvpn/easy-rsa/keys/dh2048.pem
server 10.8.0.0 255.255.255.0
# server and remote endpoints
ifconfig 10.8.0.1 10.8.0.2
# Add route to Client routing table for the OpenVPN Server
push "route 10.8.0.1 255.255.255.255"
# Add route to Client routing table for the OpenVPN Subnet
push "route 10.8.0.0 255.255.255.0"
# your local subnet
push "route 192.168.0.104 255.255.255.255" # SWAP THE IP NUMBER WITH YOUR RASPBERRY PI IP ADD$
# Set primary domain name server address to the SOHO Router
# If your router does not do DNS, you can use Google DNS 8.8.8.8
push "dhcp-option DNS 192.168.0.1" # This should match your router's IP address.
push "dhcp-option DNS 10.8.0.1"
#push "dhcp-option DNS 8.8.8.8"

# Override the Client default gateway by using 0.0.0.0/1 and
# 128.0.0.0/1 rather than 0.0.0.0/0. This has the benefit of
# overriding but not wiping out the original default gateway.
push "redirect-gateway def1"
client-to-client
#duplicate-cn
keepalive 10 120
tls-auth /etc/openvpn/easy-rsa/keys/ta.key 0
cipher AES-256-CBC
comp-lzo
user nobody
group nogroup
persist-key
persist-tun
status /var/log/openvpn-status.log 20
log /var/log/openvpn.log
verb 1

Here is the ifconfig file:
ifconfig
eth0      Link encap:Ethernet  HWaddr XX:XX:XX:XX:XX:XX
          inet addr:192.168.0.104  Bcast:192.168.0.255  Mask:255.255.255.0
          inet6 addr: XXXX:XXXX:XXX:XXXX:XXXX:XXXX:XXXX:XXXX/64 Scope:Global
          inet6 addr: XXXX::XXXX:XXXX:XXXX:XXXX/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:81566 errors:0 dropped:11 overruns:0 frame:0
          TX packets:28968 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:13718271 (13.0 MiB)  TX bytes:8256196 (7.8 MiB)

lo        Link encap:Local Loopback
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:40010 errors:0 dropped:0 overruns:0 frame:0
          TX packets:40010 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1
          RX bytes:2950224 (2.8 MiB)  TX bytes:2950224 (2.8 MiB)

wlan0     Link encap:Ethernet  HWaddr XX:XX:XX:XX:XX:XX
          inet6 addr: XXXX::XXXX::XXXX:XXXX:XXXX/64 Scope:Link
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

**Comment by technicalpyro** [comment]
https://gist.github.com/technicalpyro/5c42f839ee33a5e91321a8dc6aa9399e that is my current setup note that it is for passing only DNS trhough the VPN to disable that setting remove the comment on the line `#push "redirect-gateway def1 bypass-dhcp"`

**Comment by Clubguy** [comment]
The only OpenVPN config which I could get working was here and that was before I installed Pi-hole and broke it:
http://itblogsec.com/build-own-openvpn-server-by-using-raspberry-pi-12/
There is link to a second article at the end of the first one.  

I tried upgrdman's directions 12 times, I tried Starship  Engineer's directions, I tried Darren Kitchen's directions, I tried following the official OpenVPN instructions from OpenVPN.com and none of the worked.  I might install it on a second device and run two Raspberry Pi's to get what I need.
